cddl/identity-triple-record.cddl

```diff
@@ -1,4 +1,6 @@
 identity-triple-record = [
-  environment-map
-  [ + $crypto-key-type-choice ]
+  environment: environment-map
+  ? mkey: $measured-element-type-choice
```

Comment:
This optional mkey is a parsing annoyance. Can we move the non-environment fields to a map? Backward compatibility can just select between array and map.

Reply:
> This optional mkey is a parsing annoyance. Can we move the non-environment fields to a map? Backward compatibility can just select between array and map.

Can you suggest what the map might look like?

Reply:
I want to first know why we need these additions to begin with. mkeys are meant for measured components of an attested environment, so there should not be different attestation keys at the mkey level. Why do we need authorities when attestation keys should be accepted based an trust anchors?

Reply:
> there should not be different attestation keys at the mkey level. Why do we need authorities when attestation keys should be accepted based an trust anchors?

The mkey and authority fields (along with the other fields) are condition values used for searching / matching ACS entries. The matched entry defines the context for the addition entry which is to add an endorsement under verifier authority for the subset of keys that satisfy verification checks. The addition ACS entry is otherwise identical to the matched ACS entry. Since the current ECT structure has mkey/element it makes sense to enable searching based on the mkey/element.

Reply:
okay, then I'd suggest the following:

```
identity-triple = identity-triple-record / identity-triple-map
identity-triple-map = {
  &(environment: 0) => environment-map
  ? &(mkey: 1) => $measured-element-type-choice
  &(key-list: 2) => [ + $crypto-key-type-choice ]
  ? &(authority-list: 3) => [ + $crypto-key-type-choice ]
}
```
and not change identity-triple-record.

Reply:
What is wrong with the current proposal?

Reply:
Suppose $measured-element-type-choice is a list of tagged bytes that get misinterpreted as crypto-key-type-choice. Do you have mkey,key-list or key-list,authority-list?

Reply:
The triples map defines the identity triple as an array of `identity-triple-record`. All of the triples are defined in terms of records (and not maps). This was discussed at length.

If the new fields are placed at the end of the existing fields then the ambiguity goes away.
```
? &(identity-triples: 2) =>
    [ + identity-triple-record ]

identity-triple-record = [
  environment: environment-map
  key-list: [ + $crypto-key-type-choice ]
  ? mkey: $measured-element-type-choice
  ? authority-list: [ + $crypto-key-type-choice ]
]
```
If mkey is omitted then the second list of crypto-key-type-choice is the authority-list. If mkey is included then it isn't confused with a key list.

Reply:
I don't think the reordering avoids the ambiguity. I think you'd need to require that measured-element-type-choice must not be an array.